packages/big-design/src/components/List/Item/Item.tsx

```diff
@@ -1,32 +1,193 @@
-import React, { forwardRef, LiHTMLAttributes, memo, Ref } from 'react';
+import { CheckIcon } from '@bigcommerce/big-design-icons';
+import { UseSelectPropGetters } from 'downshift';
+import React, { cloneElement, forwardRef, isValidElement, LiHTMLAttributes, memo, Ref } from 'react';
 
-import { StyledListItem } from './styled';
+import { typedMemo } from '../../../utils';
+import { Checkbox } from '../../Checkbox';
+import { DropdownItem, DropdownLinkItem } from '../../Dropdown';
+import { Flex, FlexItem } from '../../Flex';
+import { SelectAction, SelectOption } from '../../Select';
+import { Tooltip } from '../../Tooltip';
+import { Small } from '../../Typography';
 
-export interface ListItemProps extends LiHTMLAttributes<HTMLLIElement> {
+import { StyledLink, StyledListItem } from './styled';
+
+export interface ListItemProps<T> extends LiHTMLAttributes<HTMLLIElement> {
   actionType?: 'normal' | 'destructive';
   autoWidth?: boolean;
   disabled?: boolean;
+  index: number;
   isAction?: boolean;
+  isChecked?: boolean;
   isHighlighted: boolean;
-  isSelected: boolean;
+  isSelected?: boolean;
+  item: DropdownItem | DropdownLinkItem | SelectOption<T> | SelectAction;
+  getItemProps: UseSelectPropGetters<any>['getItemProps'];
+  addItem?(item: SelectOption<T>): void;
+  removeItem?(item: SelectOption<T>): void;
 }
 
 interface PrivateProps {
   forwardedRef: Ref<HTMLLIElement>;
 }
 
-const StyleableListItem: React.FC<ListItemProps & PrivateProps> = ({
-  autoWidth = false,
-  actionType = 'normal' as 'normal',
-  forwardedRef,
-  isAction = false,
-  ...props
-}) => {
+type Items = DropdownItem | DropdownLinkItem | SelectOption<unknown> | SelectAction;
```

Comment:
I think you can add a generic to `Items`
```suggestion
type Items<T> = DropdownItem | DropdownLinkItem | SelectOption<T> | SelectAction;
```